test/e2e/specs/wp-calypso-gutenberg-upgrade-spec.js

```diff
@@ -0,0 +1,202 @@
+import { times } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import LoginFlow from '../lib/flows/login-flow.js';
+
+import GutenbergEditorComponent from '../lib/gutenberg/gutenberg-editor-component';
+
+import * as driverManager from '../lib/driver-manager.js';
+import * as dataHelper from '../lib/data-helper.js';
+import * as mediaHelper from '../lib/media-helper';
+
+const mochaTimeOut = config.get( 'mochaTimeoutMS' );
+const startBrowserTimeoutMS = config.get( 'startBrowserTimeoutMS' );
+const screenSize = driverManager.currentScreenSize();
+const host = dataHelper.getJetpackHost();
+
+let driver;
+let currentGutenbergBlocksCode;
+let galleryImages;
+
+before( async function () {
+	this.timeout( startBrowserTimeoutMS );
+	driver = await driverManager.startBrowser();
+	this.loginFlow = new LoginFlow( driver, 'gutenbergUpgradeUser' );
+	galleryImages = times( 5, () => mediaHelper.createFile() );
+} );
+
+describe( `[${ host }] Calypso Gutenberg Editor: Pages (${ screenSize })`, function () {
+	this.timeout( mochaTimeOut );
+
+	// I'm still not sure about the structure and description here, will review it soon
+	describe( 'Smoke test popular blocks: @parallel', function () {
+		[
+			'e2egbupgradehever',
+			'e2egbupgradeshawburn',
+			'e2egbupgrademorden',
+			'e2egbupgradeexford',
+			'e2egbupgrademayland',
+		].forEach( ( siteName ) => {
+			step( `Test popular blocks in ${ siteName }`, async function () {
+				return await this.loginFlow.loginAndStartNewPost( `${ siteName }.wordpress.com`, true );
+			} );
+
+			step( 'Popular blocks can be inserted and are shown without errors', async function () {
+				const contactEmail = 'testing@automattic.com';
+				const subject = "Let's work together";
+
+				const gEditorComponent = await GutenbergEditorComponent.Expect( driver );
+
+				// jetpack/tiled-gallery
+				const tiledGallery = await gEditorComponent.insertTiledGallery();
+				await tiledGallery.uploadImages( galleryImages );
+				// TODO When passing the block markup to the edge site, it might worth verifying if the images have loaded... (how?)
+				// FWIW, I can verify by looking that the images do load across sites even if the gallery was setup in a different wpcom site.
+				const tiledGalleryDisplayedInEditor = await gEditorComponent.tiledGalleryDisplayedInEditor();
+				assert.strictEqual(
+					tiledGalleryDisplayedInEditor,
+					true,
+					'The tiled gallery block is not displayed in the editor'
+				);
+
+				// jetpack/contact-form
+				await gEditorComponent.insertContactForm( contactEmail, subject );
+
+				const contactFormDisplayedInEditor = await gEditorComponent.contactFormDisplayedInEditor();
+				assert.strictEqual(
+					contactFormDisplayedInEditor,
+					true,
+					'The contact form is not displayed in the editor'
+				);
+
+				// jetpack/layout-grid
+				await gEditorComponent.insertLayoutGrid();
+				const layoutGridDisplayedInEditor = await gEditorComponent.layoutGridDisplayedInEditor();
```

Comment:
There's quite a bit of repetition here and I don't like it. I'll see how to dry it up later.